source/extensions/listener_managers/listener_manager/active_tcp_listener.cc

```diff
@@ -101,6 +101,12 @@ void ActiveTcpListener::onReject(RejectCause cause) {
 void ActiveTcpListener::onAcceptWorker(Network::ConnectionSocketPtr&& socket,
                                        bool hand_off_restored_destination_connections,
                                        bool rebalanced) {
+  // Get Round Trip Time
+  absl::optional<std::chrono::milliseconds> t = socket->lastRoundTripTime();
```

Comment:
I think this block should move to `Envoy::Http::FilterManager::onStreamComplete()`, which has access to the `Network::Connection`.

Reply:
Why move? In this case RTT will only be available for HTTP requests that end up creating streams. It will not cover the case where stream was not created, i.e. due to a protocol error. Or TCP to CONNECT case.

I think it makes sense to **add** it in the place you are suggesting, but still keeping the code for getting the initial measurement from TCP handshake. WDYT?

Reply:
Added to conn_manager_impl right before stream completes.